packages/dynamo-store/source/src/DynamoStoreSource.mts

```diff
@@ -117,7 +112,7 @@ namespace Impl {
         // TOCONSIDER revise logic to share session key etc to rule this out
         const sliceFrom = span.i - Number(items[0].index)
         const events = items.slice(sliceFrom, sliceFrom + span.c.length)
-        for (const event of events) result.push([StreamName.parse(span.p), event])
+        for (const event of events) result.push([StreamName.parse(span.p), inflate(event)])
```

Comment:
hm if the inflate can be deferred, it might save memory consumption, memory churn and CPU if processing can filter based on event type or index

Reply:
Actually, WTH, I've done it, see implementation of `inflate`